src/StoreApi/Utilities/OrderController.php

```diff
@@ -434,6 +437,23 @@ public function default_order_status() {
 		return 'checkout-draft';
 	}
 
+	/**
+	 * Passes the correct base for local pick orders
+	 *
+	 * @todo: Remove custom local pickup handling once WooCommerce 6.8.0 is the minimum version.
+	 *
+	 * @param array $location Taxes location.
+	 * @return array updated location that accounts for local pickup.
+	 */
+	public function handle_local_pickup_taxes( $location ) {
+		$customer = wc()->customer;
+
+		if ( ! empty( $customer ) ) {
+			return $customer->get_taxable_address();
+		}
+
+		return $location;
+	}
```

Comment:
`get_tax_location` has a filter in which we're using. In this case, we check if we have a user (we always do at the submit point) and get the address from there (which accounts for local pickup).